### PR TITLE
[WIP][ADD] web_text_count_field

### PR DESCRIPTION
**Pull request by asaunier**
Field ``TextCount``, combined to widget ``text_count``, is displayed as a usual text field with a character counter next to it.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Can't you do it only client side for not needing `TextCount` field?

File: web_text_count_field/models/fields.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+
+class TextCount(fields.Text):
+
+    def __init__(self, string, size=128, **kwargs):
+        assert size, "Size is mandatory in text_count field"
+        super().__init__(string=string, size=size, **kwargs)
+
+
+fields.TextCount = TextCount
```
Comment (simahawk) [inline web_text_count_field/models/fields.py]: do we really need a new type of field?

File: web_text_count_field/static/src/js/fields_text_count.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    var registry = require('web.field_registry');
+    var FieldText = require('web.FieldText');
+
+    var FieldTextCount = FieldText.extend({
+        template: 'FieldTextCount',
+
+        start: function () {
+            this._super();
+            var $textarea = this.$element.find('textarea')[0];
+            $textarea.addEventListener("keyup", this.count_char);
```
Comment (simahawk) [inline web_text_count_field/static/src/js/fields_text_count.js:13]: use `events` and remove `start` override

File: web_text_count_field/static/src/js/fields_text_count.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        },
+
+        count_char: function (e) {
+            var len =  this.value.length;
+            var counter = $(this.parentNode).find('span')[0];
```
Comment (simahawk) [inline web_text_count_field/static/src/js/fields_text_count.js:18]: give the counter element a class and find sibling element aside the main element

File: web_text_count_field/static/src/js/fields_text_count.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            counter.innerHTML = len;
+        },
+
+        update_dom: function () {
```
Comment (simahawk) [inline web_text_count_field/static/src/js/fields_text_count.js:22]: do we need this?
also, don't we lock editing when max size is reached?
In that case we should use the `maxlenght` property

**Review by simahawk — CHANGES_REQUESTED** [review]

**Comment by asaunier** [comment]
@pedrobaeza @simahawk I talked to @yvaucher, who wrote the module for a v7 project. We could indeed remove the python field and update the JS code as suggested.

File: web_text_count_field/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2019 Camptocamp
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Web Textarea with Counter",
+    "category": "Web",
+    "version": "12.0.1.0.0",
+    "author": "Camptocamp SA, Odoo Community Association (OCA)",
```
Comment (nikul-serpentcs) [inline web_text_count_field/__manifest__.py:7]: @asaunier Add `website` key and URL: `https://github.com/OCA/web`

**Review by nikul-serpentcs — CHANGES_REQUESTED** [review]
Minor change

File: web_text_count_field/views/templates.xml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+    <template id="assets_backend" inherit_id="web.assets_backend">
+        <xpath expr="." position="inside">
+            <link href="/web_text_count_field/static/src/scss/field_text_count.scss" rel="stylesheet" type="text/css"/>
```
Comment (i-vyshnevska) [inline web_text_count_field/views/templates.xml:5]:
```suggestion
            <link href="/web_text_count_field/static/src/scss/field_text_count.scss" rel="stylesheet" type="text/scss"/>
```

File: web_text_count_field/static/src/xml/fields_text_count.xml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates>
+
+  <t t-name="FieldTextCount" t-extend="FieldText">
```
Comment (i-vyshnevska) [inline web_text_count_field/static/src/xml/fields_text_count.xml:4]: template for "FieldText" not existing anymore

File: web_text_count_field/static/src/js/fields_text_count.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
+odoo.define('text_count', function(require) {
+    "use strict";
+
+    var registry = require('web.field_registry');
+    var FieldText = require('web.FieldText');
```
Comment (i-vyshnevska) [inline web_text_count_field/static/src/js/fields_text_count.js:5]:
```suggestion
    var FieldText = require('web.basic_fields').FieldText;
```

**Review by i-vyshnevska — CHANGES_REQUESTED** [review]

**Review by tarteo — APPROVED** [review]
LGTM besides the comments that are already made. I'd rather see this as an option on the existing text widget, but this is also fine for me.

**Comment by pedrobaeza** [comment]
Superseeded by #1357

**Pull request closed without merging**